src/lib.rs

```diff
@@ -61,69 +76,253 @@ impl InlinerBuilder {
         self
     }
 
-    /// Configures whether unexpanded modules (due to a missing file) will lead
-    /// to an `Err` return value or not.
+    /// Configures whether unexpanded modules (due to missing files or invalid Rust source code)
+    /// will lead to an `Err` return value or not.
     ///
     /// Default: `false`.
-    pub fn error_not_found(&mut self, error_not_found: bool) -> &mut Self {
-        self.error_not_found = error_not_found;
+    pub fn track_errors(&mut self, track_errors: bool) -> &mut Self {
+        self.track_errors = track_errors;
         self
     }
 
     /// Parse the source code in `src_file` and return a `syn::File` that has all modules
     /// recursively inlined.
-    pub fn parse_and_inline_modules(&self, src_file: &std::path::Path) -> Result<syn::File, Error> {
+    pub fn parse_and_inline_modules(&self, src_file: &Path) -> Result<syn::File, Error> {
         self.parse_internal(src_file, FsResolver::default())
     }
 
     fn parse_internal<R: FileResolver + Clone>(
         &self,
-        src_file: &std::path::Path,
+        src_file: &Path,
         resolver: R,
     ) -> Result<syn::File, Error> {
-        let mut errors = if self.error_not_found {
+        let mut errors = if self.track_errors {
             Some(vec![])
         } else {
             None
         };
         let result =
             Visitor::<R>::with_resolver(src_file, self.root, errors.as_mut(), Cow::Owned(resolver))
-                .visit();
+                .visit()
+                .map_err(Error::Root)?;
         match errors {
             Some(ref errors) if errors.is_empty() => Ok(result),
             None => Ok(result),
-            Some(errors) => Err(Error::NotFound(errors)),
+            Some(errors) => Err(Error::Inline(IncompleteInlining::new(result, errors))),
         }
     }
 }
 
-/// A source location
-#[derive(Debug, Clone, PartialEq, Eq)]
-pub struct SourceLocation {
-    pub path: PathBuf,
-    pub line: usize,
-    pub column: usize,
-    _private: (),
+/// An error that was encountered while reading, parsing or inlining a module.
+#[derive(Debug)]
+pub enum Error {
+    /// An error happened while reading or parsing the initial file.
+    ///
+    /// For example, the initial file wasn't found or wasn't a valid Rust file.
+    Root(ErrorKind),
```

Comment:
My concern with `Root` was just that `InlinerBuilder` has the [`root` method](https://docs.rs/syn-inline-mod/0.3.0/syn_inline_mod/struct.InlinerBuilder.html#method.root) which meaning conflicts with this a bit. That's why I went with `Initial`.

Reply:
Good point. I'm trying - and failing - to remember if `root` was something that I meant to make public, or if that was for internal use and it shouldn't have leaked.

Reply:
Naming issue no longer exists.